lib/agent0/agent0/hyperdrive/crash_report/crash_report.py

```diff
@@ -81,6 +90,79 @@ def setup_hyperdrive_crash_report_logging(log_format_string: str | None = None)
     )
 
 
+def build_crash_trade_result(
+    exception: Exception,
+    agent: HyperdriveAgent,
+    trade_object: types.Trade[HyperdriveMarketAction],
+    hyperdrive: HyperdriveInterface,
+) -> TradeResult:
+    """Build the trade result object when a crash happens.
+
+    Arguments
+    ---------
+    exception : Exception
+        The exception that was thrown
+    """
+    trade_result = TradeResult(
+        status=TradeStatus.FAIL,
+        agent=agent,
+        trade_object=trade_object,
+    )
+
+    # We log pool config and pool info here
+    # However, this is a best effort attempt to get this information
+    # due to async conditions. If debugging this crash, ensure the agent is running
+    # in isolation and doing one trade per call.
+
+    # We get the underlying contract info and convert them to human readable versions
+    # Dispite these being protected variables, we need low level access for crash reporting
+    # TODO we likely should call underlying web3 commands here to prevent race conditions
```

Comment:
this todo might be resolved by the proposed hyperdrive interface in #1022 fwiw

Reply:
<3